inference-engine/src/mkldnn_plugin/nodes/mkldnn_reduce_node.cpp

```diff
@@ -0,0 +1,1350 @@
+                    uni_vbroadcastss(vmm_aux, ptr[reg_aux]);
+                    uni_vpxor(vmm_dst, vmm_dst, vmm_dst);
+                    break;
+                case Reduce::L2:
+                case Reduce::LogSum:
+                case Reduce::LogSumExp:
+                case Reduce::Mean:
+                case Reduce::Sum:
+                case Reduce::SumSquare:
+                    uni_vpxor(vmm_dst, vmm_dst, vmm_dst);
+                    break;
+                case Reduce::Max:
+                    if (jcp_.dst_dt == memory::f32)
+                        mov(reg_aux, ptr[reg_params + GET_OFF(float_min_aux)]);
+                    else
+                        mov(reg_aux, ptr[reg_params + GET_OFF(int32_min_aux)]);
+                    uni_vbroadcastss(vmm_dst, ptr[reg_aux]);
+                    break;
+                case Reduce::Min:
+                    if (jcp_.dst_dt == memory::f32)
+                        mov(reg_aux, ptr[reg_params + GET_OFF(float_max_aux)]);
+                    else
+                        mov(reg_aux, ptr[reg_params + GET_OFF(int32_max_aux)]);
+                    uni_vbroadcastss(vmm_dst, ptr[reg_aux]);
+                    break;
+                case Reduce::Or:
+                    uni_vpxor(vmm_dst, vmm_dst, vmm_dst);
+                    break;
+                default:
+                    THROW_IE_EXCEPTION << "Unknown reduce mode.";
+            }
+            //reduce
+            reduce_loop();
+            if (jcp_.reduce_mode == Reduce::Or && isa != avx512_common) {
+                vcmpneqps(vmm_dst, vmm_dst, vmm_zero);
+                uni_vandps(vmm_dst, vmm_dst, vmm_aux);
+            }
+            //store
+            horiz_reduce_store(jcp_.dst_dt);
+        } else { //blocked
+            //load
+            if (jcp_.reduce_mode == Reduce::L1) {
+                mov(reg_aux, ptr[reg_params + GET_OFF(float_abs_aux)]);
+                uni_vbroadcastss(vmm_aux, ptr[reg_aux]);
+            }
+            load_vector(vmm_dst, ptr[reg_dst], jcp_.dst_dt);
+            if (isa == cpu::sse42)
+                load_vector(vmm_dst_aux, ptr[reg_dst + 4 * jcp_.dst_data_size], jcp_.dst_dt);
+            //reduce
+            reduce_loop();
+            if (jcp_.reduce_mode == Reduce::Or && isa != avx512_common) {
+                vcmpneqps(vmm_dst, vmm_dst, vmm_zero);
+                uni_vandps(vmm_dst, vmm_dst, vmm_aux);
+                if (isa == cpu::sse42) {
+                    vcmpneqps(vmm_dst_aux, vmm_dst_aux, vmm_zero);
+                    uni_vandps(vmm_dst_aux, vmm_dst_aux, vmm_aux);
+                }
+            }
+            //store
+            store_vector(ptr[reg_dst], vmm_dst, jcp_.dst_dt);
+            if (isa == cpu::sse42)
+                store_vector(ptr[reg_dst + 4 * jcp_.dst_data_size], vmm_dst_aux, jcp_.dst_dt);
+        }
+
+        this->postamble();
+
+        exp_injector->prepare_table();
```

Comment:
Looks like we can all this only for specific algorithm, not for all.

Reply:
Fixed. Thanks Dmitry!